graphql/database/missions/utils.js

```diff
@@ -0,0 +1,154 @@
+/* eslint-disable no-nested-ternary */
+/* eslint-disable no-param-reassign */
+import UserModel from '../users/UserModel'
+import {
+  getPermissionsOverride,
+  MISSIONS_OVERRIDE,
+} from '../../utils/permissions-overrides'
+
+const override = getPermissionsOverride(MISSIONS_OVERRIDE)
+export const buildSquadMemberDataFromMissionDoc = missionDoc => {
+  const {
+    acceptedSquadMembers,
+    pendingSquadMembersExisting,
+    rejectedSquadMembers,
+    pendingSquadMembersEmailInvite,
+  } = missionDoc
+  const squadMemberDataFromMissionDocAsDictionary = {}
+  acceptedSquadMembers.forEach(acceptedUserId => {
+    squadMemberDataFromMissionDocAsDictionary[acceptedUserId] = {
+      userId: acceptedUserId,
+      status: 'accepted',
+    }
+  })
+  pendingSquadMembersExisting.forEach(pendingUserId => {
+    squadMemberDataFromMissionDocAsDictionary[pendingUserId] = {
+      status: 'pending',
+      userId: pendingUserId,
+    }
+  })
+  rejectedSquadMembers.forEach(rejectedUserId => {
+    squadMemberDataFromMissionDocAsDictionary[rejectedUserId] = {
+      status: 'rejected',
+      userId: rejectedUserId,
+    }
+  })
+  pendingSquadMembersEmailInvite.forEach(email => {
+    squadMemberDataFromMissionDocAsDictionary[email] = {
+      status: 'rejected',
+      invitedEmail: email,
+    }
+  })
+  return squadMemberDataFromMissionDocAsDictionary
```

Comment:
Looks like it would be cleaner to just have this be an array? I think you just end up reducing it below anyway.

Reply:
if it was an array i'd have to run indexOf on each user to look up username/email.  this function isn't my favorite code since i still have to turn it into an array but I think it's like 3n instead of n^2.  Either way, groups won't have high user counts so 🤷

Reply:
@amaliwanag2 Ah ok, gotcha.